File: ndannamo_project/backend/src/main/java/com/example/backend/model/Activity.java
package com.example.backend.model;


import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.*;
import java.time.LocalTime;

import io.micrometer.common.lang.Nullable;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name="activities")
public class Activity extends Event {

    // Per EventDTO
    private EventType type = EventType.ACTIVITY;

    private String name;
    private LocalTime startTime;

    @Nullable
    private LocalTime endTime = null;       // non e' detto che ci sia

    private String address;
    private String info;
}

File: source/frontend/src/models/Activity.js
import ActivityAndTravel from "./ActivityAndTravel";

export default class Activity extends ActivityAndTravel {
    constructor(id, tripId, place, date, address, startTime, endTime, info, name) {
        super(id, tripId, place, date, address, startTime, endTime, info);
        this.name = name;

    }
}
